ffi/src/lib.rs

```diff
@@ -159,18 +275,35 @@ pub unsafe extern "C" fn store_transact(store: *mut Store, transaction: *const c
     Box::into_raw(Box::new(result.into()))
 }
 
+/// Fetches the `tx_id` for the given `TxReport`.
+///
+/// # Safety
```

Comment:
Assuming you mean "points to a Rust TxReport" by "not dangling", I don't think these safety blocks help in this case.  Garbage in, garbage out -- and there's nothing to be done our side if the consumer is just wrong.